src/test/runtimeChoiceSelector.spec.ts

```diff
@@ -0,0 +1,102 @@
+import * as sinon from "sinon";
+import { expect } from "chai";
+import * as chai from "chai";
+import * as utils from "../utils";
+import * as runtime from "../runtimeChoiceSelector";
+import * as FirebaseError from "../error";
+// have to require this because no @types/cjson available
+// tslint:disable-next-line
+const cjson = require("cjson");
+
+describe("getRuntimeName", () => {
+  it("should properly convert raw runtime to human friendly runtime", () => {
+    expect(runtime.getHumanFriendlyRuntimeName("nodejs6")).to.contain("Node.js");
+  });
+});
+
+describe("getRuntimeChoice", () => {
+  const sandbox: sinon.SinonSandbox = sinon.createSandbox();
+  let cjsonStub: sinon.SinonStub;
+  let warningSpy: sinon.SinonSpy;
+
+  beforeEach(() => {
+    cjsonStub = sandbox.stub(cjson, "load");
+    warningSpy = sandbox.spy(utils, "logWarning");
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  it("should return node 6 if package.json engines field is set to node 6 and print warning", () => {
+    cjsonStub.returns({ engines: { node: "6" }, dependencies: { "firebase-functions": "2.0.0" } });
+
+    expect(runtime.getRuntimeChoice("path/to/source")).to.deep.equal("nodejs6");
+    expect(warningSpy).calledWith(runtime.DEPRECATION_WARNING_MSG);
```

Comment:
Alright, it's been a while since I've done this, but here's a different way to test this than copying/exporting entire strings and making sure they match.

Each of the deprecation warnings has some specific text in them. You can use a regular expression to match those keywords (around all the other content). To do this with `calledWith`, sinon provides [matchers](https://sinonjs.org/releases/latest/matchers/). It would make this check look like:

```suggestion
    expect(warningSpy).calledWith(sinon.match(/Node 6 runtime.+deprecated/));
```

How does that feel?

Reply:
Leaving the constants in as per our other offline conversation.

Reply:
👍